usecases/replica/coordinator.go

```diff
@@ -159,6 +159,14 @@ func (c *coordinator[T]) Push(ctx context.Context,
 	ask readyOp,
 	com commitOp[T],
 ) (<-chan _Result[T], int, error) {
+	// If ConsistencyLevel is not All then replace the context with a background context. This is necessary because
+	// otherwise we will return to the client once we have one or quorum on the broadcast. Depending on the speed of the
+	// replication this might lead to the client closing the request and the original context getting canceled.
+	// That cancelation would lead to the replication requests to the other nodes to get cancelled as-well and
+	// replication failing, which results in dataloss.
+	if cl != All {
```

Comment:
**thought(blocking)**: This will make the request not being cancellable. I don't believe it can result in data loss. 
The `ctx` is used solely for broadcasting replicaiton request to replicas and doesn't perform any write. Therefore it is safe to cancel it. The actually write occurs when commitAll(), which uses a background context instead of the provided `ctx`. The root of the problem might be elsewhere

Reply:
> I don't believe it can result in data loss.
It resulted in data loss as the count of insert was off during our pipeline testing on replication (chaos pipeline replication while importing, no matter if we were killing nodes or not).

If the context gets canceled once we have ONE or QUORUM then we will not have this node in the resulting `nodeCh` because the query to it will result in an error and we will not have them commit.